### PR TITLE
Add username to request object so apps can do stuff with it

### PR DESCRIPTION
**Pull request by lucas42**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -22,6 +22,9 @@ var authenticateToken = function (res, username, hostname, token) {
 
 	if (result) {
 		debug('S3O: Authentication successful: ' + username);
+
+		// Add username to request, so apps can utilise it.
+		req.s3o_username = username;
```
Comment (matthew-andrews) [inline index.js]: don't you think `res.locals` is a better place for this sort of data?

Comment (matthew-andrews) [inline index.js]: also would make this PR a bit small.

Comment (lucas42) [inline index.js]: That could work.  I've never used res.locals before.  I'll give it a go :)

**Comment by matthew-andrews** [comment]
+1

**Comment by matthew-andrews** [comment]
i can pubilsh for you if yo8u like

**Comment by lucas42** [comment]
Yes please!

**Comment by matthew-andrews** [comment]
```sh
$ npm publish
+ s3o-middleware@1.6.0

```

**Pull request merged**